src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -0,0 +1,87 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using EnsureThat;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Client
+{
+    // Todo need to be async
+
+    public class OrasClient : IOrasClient
+    {
+        private readonly string _imageReference;
+
+        private readonly string _workingImageLayerFolder;
+
+        public OrasClient(string imageReference, string workingFolder)
```

Comment:
>imageReference [](start = 33, length = 14)

maybe move it into pull & push

Reply:
Maybe one client corresponds to one imageReference?